src/client/sw.js

```diff
@@ -3,13 +3,37 @@ let version = 'VERSION'
 workbox.setConfig({ debug: SW_DEBUG })
 workbox.skipWaiting()
 workbox.clientsClaim()
-workbox.core.setCacheNameDetails({ prefix: 'hn-pwa', suffix: `v2.${version}`, precache: 'precache', runtime: 'runtime' })
+
+// for precache we dont want to set versioning since every new version will start with empty cache
+// workbox already has a mechanism to update resources in precache based on their revision
+// this way only the resources that changed since last build will be replaced in cache
+workbox.core.setCacheNameDetails({ prefix: 'hn-pwa', precache: 'precache', runtime: `runtime-v2.${version}` })
 
 // Application
 workbox.routing.registerRoute(/^\/images\/.*/, workbox.strategies.cacheFirst())
-workbox.routing.registerRoute('/manifest.json', workbox.strategies.staleWhileRevalidate())
```

Comment:
removing manifest caching here because it is already cached in static assets and `precache` phase